src/main/reactjs/src/tests/cypress/integration/public_translator_filters.spec.ts

```diff
@@ -0,0 +1,33 @@
+import { APIEndpoints } from 'enums/api';
+import { onPublicTranslatorFilters } from 'tests/cypress/support/page-objects/publicTranslatorFilters';
+import { onPublicTranslatorsListing } from 'tests/cypress/support/page-objects/publicTranslatorsListing';
+import { runWithIntercept } from 'tests/cypress/support/utils/api';
+
+beforeEach(() => {
+  runWithIntercept(
+    APIEndpoints.PublicTranslator,
+    { fixture: 'public_translators_50.json' },
+    () => cy.openPublicHomePage()
+  );
+});
+
+describe('PublicTranslatorFilters', () => {
+  it('should allow filtering results by language pair, name and town', () => {
+    onPublicTranslatorFilters.filterByLanguagePair('suomi', 'ruotsi');
+    onPublicTranslatorsListing.expectTranslatorsCount(27);
+
+    onPublicTranslatorFilters.filterByTown('Helsinki');
+    onPublicTranslatorsListing.expectTranslatorsCount(2);
+
+    onPublicTranslatorFilters.filterByName('aaltonen a');
+    onPublicTranslatorsListing.expectTranslatorsCount(1);
+  });
+
+  it('reset button should clear filters and listed translators', () => {
```

Comment:
Olisiko seuraava lause parempi? 

--> should clear filters and listed translators when the reset button is clicked.

Silloin sitä on helpompi lukea it osan kanssa :).

Reply:
Jep, muutan.